homework/Truong_Duong/lesson_3/215_Kth_Largest_Element.js

```diff
@@ -0,0 +1,42 @@
+// Solution for https://leetcode.com/problems/kth-largest-element-in-an-array/
+
+/**
+ * @param {number[]} nums
+ * @param {number} k
+ * @return {number}
+ */
+
+function swap(arr, i, j){
+  [arr[i], arr[j]] = [arr[j], arr[i]];
+}
+
+function partition(nums, left, right) {
+  boundary = left;
+  pivot = nums[right];
+
+  for (let i = left; i < right; i++) {
+    if(nums[i] < pivot) {
+      swap(nums, i, boundary);
+      boundary++;
+    }
+  }
+  swap(nums, right, boundary);
+  return boundary;
+}
+
+function findKthLargestUtil(nums, pos, left, right) {
+  const pivotIndex = partition(nums, left, right);
+
+  if (pivotIndex === pos) {
+    return nums[pos];
+  } else if (pivotIndex > pos) {
+    return findKthLargestUtil(nums, pos, left, boundary -1);
+  } else {
+    return findKthLargestUtil(nums, pos, boundary + 1, right);
+  }
+}
```

Comment:
Just a suggestion, for me if there is `return` statement inside `if` block, i will consider not using `else` or `else if`. This block code can be more concise as below:
```suggestion
  if (pivotIndex === pos) {
    return nums[pos];
  } 
  
  if (pivotIndex > pos) {
    return findKthLargestUtil(nums, pos, left, boundary -1);
  }
  
   return findKthLargestUtil(nums, pos, boundary + 1, right);
}
```

Reply:
I liked your approach, really appreciate your comment 🙇

Reply:
Glad to hear it helps you!